simulator/infra/macro.h

```diff
@@ -173,6 +173,13 @@ static constexpr T ones_ls(const T& value, size_t shamt)
     return ~(~value << shamt);
 }
 
+/*Circular left shift*/
+template<typename T>
+static constexpr T circ_ls(const T& value, size_t shamt)
+{
+    return (value << shamt) | (value >> ((bitwidth<T> - shamt) & (bitwidth<T> - 1)));
```

Comment:
Is `& (bitwidth<T> - 1)` redundant?

Reply:
What happens in case of `shamt == 0`?
I propose to add unit tests for that function (`infra/t/unit_test.cpp').